MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m

```diff
@@ -189,6 +188,17 @@ - (void)checkLatestRelease:(NSString *)updateToken releaseHash:(NSString *)relea
   // Check if it's okay to check for updates.
   if ([self checkForUpdatesAllowed]) {
 
+    // Use persisted mandatory update while network is down.
+    if ([MS_Reachability reachabilityForInternetConnection].currentReachabilityStatus == NotReachable) {
+      self.mandatoryRelease =
+          [[MSReleaseDetails alloc] initWithDictionary:[MS_USER_DEFAULTS objectForKey:kMSMandatoryReleaseKey]];
+      if (self.mandatoryRelease && ![self handleUpdate:self.mandatoryRelease]) {
+
+        // This release is no more a candidate for update, deleting it.
+        [self deleteMandatoryRelease];
+      }
+    }
+
     // Check if sender is still waiting for a response of the previous request.
     if (self.sender == nil) {
```

Comment:
We still enqueue the request to the sender so that when the network will be back then it will automatically fetch the new release.